TestBeatKibanaRef is flaky

```
=== RUN   TestBeatKibanaRef/Verify_dashboards_installed
Retries (5m0s timeout): ..................................................................................................
    utils.go:84: 
        	Error Trace:	utils.go:84
        	Error:      	Received unexpected error:
        	            	expected  Filebeat dashboard [true], found dashboards [false]
        	Test:       	TestBeatKibanaRef/Verify_dashboards_installed
{"log.level":"error","@timestamp":"2020-09-01T17:50:04.214Z","message":"stopping early","service.version":"0.0.0-SNAPSHOT+00000000","service.type":"eck","ecs.version":"1.4.0","error":"test failure","error.stack_trace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128\ngithub.com/elastic/cloud-on-k8s/test/e2e/test.StepList.RunSequential\n\t/go/src/github.com/elastic/cloud-on-k8s/test/e2e/test/step.go:43\ngithub.com/elastic/cloud-on-k8s/test/e2e/beat.TestBeatKibanaRef\n\t/go/src/github.com/elastic/cloud-on-k8s/test/e2e/beat/setup_test.go:117\ntesting.tRunner\n\t/usr/local/go/src/testing/testing.go:1108"}
    --- FAIL: TestBeatKibanaRef/Verify_dashboards_installed (300.00s)

```

--- a/test/e2e/beat/setup_test.go
+++ b/test/e2e/beat/setup_test.go
@@ -18,6 +18,8 @@ import (
 	"github.com/elastic/cloud-on-k8s/test/e2e/test/beat"
 	"github.com/elastic/cloud-on-k8s/test/e2e/test/elasticsearch"
 	"github.com/elastic/cloud-on-k8s/test/e2e/test/kibana"
+
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
 )
 
 type kbSavedObjects struct {
@@ -140,6 +142,9 @@ func getDashboardCheck(esBuilder elasticsearch.Builder, kbBuilder kibana.Builder
 					}
 					if dashboards.Total == 0 {
 						return fmt.Errorf("expected >0 dashboards but got 0")
+					}
+					for _, db := range dashboards.SavedObjects {
+						logf.Log.Info("Found dashboard", "title", db.Attributes.Title)
 					}
 					for beat, expectDashboards := range beatToDashboardsPresence {
 						// We are exploiting the fact here that Beats dashboards follow a naming convention that starts with the

--- a/test/e2e/beat/setup_test.go
+++ b/test/e2e/beat/setup_test.go
@@ -7,7 +7,6 @@ package beat
 import (
 	"encoding/json"
 	"fmt"
-	"strings"
 	"testing"
 
 	esv1 "github.com/elastic/cloud-on-k8s/pkg/apis/elasticsearch/v1"
@@ -18,8 +17,6 @@ import (
 	"github.com/elastic/cloud-on-k8s/test/e2e/test/beat"
 	"github.com/elastic/cloud-on-k8s/test/e2e/test/elasticsearch"
 	"github.com/elastic/cloud-on-k8s/test/e2e/test/kibana"
-
-	logf "sigs.k8s.io/controller-runtime/pkg/log"
 )
 
 type kbSavedObjects struct {
@@ -29,15 +26,6 @@ type kbSavedObjects struct {
 			Title string `json:"title"`
 		} `json:"attributes"`
 	} `json:"saved_objects"`
-}
-
-func (so kbSavedObjects) HasDashboardsWithPrefix(prefix string) bool {
-	for _, obj := range so.SavedObjects {
-		if strings.HasPrefix(obj.Attributes.Title, prefix) {
-			return true
-		}
-	}
-	return false
 }
 
 func TestBeatKibanaRefWithTLSDisabled(t *testing.T) {
@@ -130,30 +118,27 @@ func getDashboardCheck(esBuilder elasticsearch.Builder, kbBuilder kibana.Builder
 						return err
 					}
 
-					body, err := kibana.DoRequest(client, kbBuilder.Kibana, password,
-						"GET", "/api/saved_objects/_find?type=dashboard", nil,
-					)
-					if err != nil {
-						return err
-					}
-					var dashboards kbSavedObjects
-					if err := json.Unmarshal(body, &dashboards); err != nil {
-						return err
-					}
-					if dashboards.Total == 0 {
-						return fmt.Errorf("expected >0 dashboards but got 0")
-					}
-					for _, db := range dashboards.SavedObjects {
-						logf.Log.Info("Found dashboard", "title", db.Attributes.Title)
-					}
 					for beat, expectDashboards := range beatToDashboardsPresence {
-						// We are exploiting the fact here that Beats dashboards follow a naming convention that starts with the
-						// name of the beat in square brackets. This test will obviously break if future versions of Beats
-						// abandon this naming convention.
-						hasDashboards := dashboards.HasDashboardsWithPrefix(fmt.Sprintf("[%s ", beat))
-						if hasDashboards != expectDashboards {
-							return fmt.Errorf("expected  %s dashboard [%v], found dashboards [%v]", beat, expectDashboards, hasDashboards)
+						// We are exploiting the fact here that Beats dashboards follow a naming convention that contains the
+						// name of the beat. This test will obviously break if future versions of Beats abandon this naming convention.
+						query := fmt.Sprintf("/api/saved_objects/_find?type=dashboard&search_fields=title&search=%s", beat)
+						body, err := kibana.DoRequest(client, kbBuilder.Kibana, password,
+							"GET", query, nil,
+						)
+						if err != nil {
+							return err
 						}
+						var dashboards kbSavedObjects
+						if err := json.Unmarshal(body, &dashboards); err != nil {
+							return err
+						}
+						if dashboards.Total == 0 && expectDashboards {
+							return fmt.Errorf("expected %s dashboards, but found none", beat)
+						}
+						if dashboards.Total != 0 && !expectDashboards {
+							return fmt.Errorf("expected no %s dashboards, but found some", beat)
+						}
+
 					}
 					return nil
 				}),

